src/pjm/parser.jl

```diff
@@ -0,0 +1,152 @@
+const DayAheadHourlyLMP_values_keys = [
+    "system_energy_price_da",
+    "total_lmp_da",
+    "congestion_price_da",
+    "marginal_loss_price_da",
+]
+
+const DayAheadHourlyLMP_meta_keys = [
+    "pnode_name",
+    "voltage",
+    "equipment",
+    "type",
+    "zone",
+    "row_is_current",
+    "version_nbr",
+]
+
+struct DayAheadHourlyLMP <: ElectricityMarket
+    Meta::DataFrame
+    SystemEnergyPricesAhead::DataFrame
+    TotalLMPDayAhead::DataFrame
+    CongestionPriceDayAhead::DataFrame
+    MarginalLossPriceDayAhead::DataFrame
+
+    function DayAheadHourlyLMP(start_date::String, start_hour_minute, end_date::String, end_hour_minute::String)
+        df = get_day_ahead_hourly_lmp(start_date, start_hour_minute, end_date, end_hour_minute)
+        df_dict, meta_df = parse_df_format(df, DayAheadHourlyLMP_values_keys, DayAheadHourlyLMP_meta_keys)
+        return new(
+            meta_df,
+            df_dict["system_energy_price_da"],
+            df_dict["total_lmp_da"],
+            df_dict["congestion_price_da"],
+            df_dict["marginal_loss_price_da"],
+        )
+    end
+end
+
+"""
+    get_day_ahead_hourly_lmp(start_date::String, end_date::String)::DataFrame
+
+Get the dataframe for the day ahead hourly lmp data from the PJM API between `start_date` and `end_date`.
+
+# Arguments
+- start_date::String: The start date of the data to be retrieved (format: "m/d/y").
+- end_date::String: The end date of the data to be retrieved (format: "m/d/y").
+- start_hour_minute::String: The start hour and minute of the data to be retrieved (format: "HH:MM").
+- end_hour_minute::String: The end hour and minute of the data to be retrieved (format: "HH:MM").
+
+# Returns
+- DataFrame: The dataframe for the day ahead hourly lmp data from the PJM API between `start_date` and `end_date`.
+"""
+function get_day_ahead_hourly_lmp(start_date::String, start_hour_minute, end_date::String, end_hour_minute)::DataFrame
+    access_key_dict = get_acess_key_headers()
+    url = get_url_day_ahead_hourly_lmp(start_date, start_hour_minute, end_date, end_hour_minute)
+    response = HTTP.get(url, headers=access_key_dict)
+
+    return CSV.read(IOBuffer(response.body), DataFrame)
```

Comment:
It is nice that we can get the data directly without downloading files. Nevertheless, it would be nice to have an option to cache the data in it's raw format.